simapp/sim_test.go

```diff
@@ -627,15 +720,47 @@ func BenchmarkInvariants(b *testing.B) {
 	}()
 
 	app := NewSimApp(logger, db, nil, true, 0)
+	exportParams := exportParamsPath != ""
 
 	// 2. Run parameterized simulation (w/o invariants)
-	_, err := simulation.SimulateFromSeed(
+	_, params, simErr := simulation.SimulateFromSeed(
 		b, ioutil.Discard, app.BaseApp, appStateFn, seed, testAndRunTxs(app),
-		[]sdk.Invariant{}, numBlocks, blockSize, commit, lean, onOperation, false,
-		app.ModuleAccountAddrs(),
+		[]sdk.Invariant{}, numBlocks, exportParamsHeight, blockSize,
+		exportParams, commit, lean, onOperation, false, app.ModuleAccountAddrs(),
 	)
-	if err != nil {
-		fmt.Println(err)
+
+	// export state and params before the simulation error is checked
+	if exportStatePath != "" {
+		fmt.Println("Exporting app state...")
+		appState, _, err := app.ExportAppStateAndValidators(false, nil)
+		if err != nil {
+			fmt.Println(err)
+			b.Fail()
+		}
+		err = ioutil.WriteFile(exportParamsPath, []byte(appState), 0644)
```

Comment:
Maybe I'm missing something?
```suggestion
		err = ioutil.WriteFile(exportStatePath, []byte(appState), 0644)
```